conda_rpms/install.py

```diff
@@ -399,6 +404,195 @@ def symlink_conda(prefix, root_dir):
     if not os.path.lexists(prefix_deactivate):
         os.symlink(root_deactivate, prefix_deactivate)
 
+# ========================== begin noarch functions =========================
+# Below are a collection of functions that are used when installing noarch
+# packages. The functions have been copied (and in some places modified) from
+# conda at commit sha be8c08c083f4d5e05b06bd2689d2cd0d410c2ffe.
+
+python_entry_point_template = dedent("""
+# -*- coding: utf-8 -*-
+import re
+import sys
+from %(module)s import %(import_name)s
+if __name__ == '__main__':
+    sys.argv[0] = re.sub(r'(-script\.pyw?|\.exe)?$', '', sys.argv[0])
+    sys.exit(%(func)s())
+""").lstrip()
+
+
+def pyc_path(py_path, python_major_minor_version):
+    """
+    Copy of conda/common/path.py:pyc_path at
+    be8c08c083f4d5e05b06bd2689d2cd0d410c2ffe.
+
+    """
+    pyver_string = python_major_minor_version.replace('.', '')
+    if pyver_string.startswith('2'):
+        return py_path + 'c'
+    else:
+        directory, py_file = split(py_path)
+        basename_root, extension = splitext(py_file)
+        pyc_file = "__pycache__/%s.cpython-%s%sc" % (basename_root, pyver_string, extension)
+        return "%s/%s" % (directory, pyc_file) if directory else pyc_file
+
+
+def get_python_noarch_target_path(source_short_path, target_site_packages_short_path):
+    """
+    Modified version of conda/common/path.py:get_python_noarch_target_path at
+    be8c08c083f4d5e05b06bd2689d2cd0d410c2ffe.
+
+    Modifications included:
+        * removed windows specific support
+
+    """
+    if source_short_path.startswith('site-packages/'):
+        sp_dir = target_site_packages_short_path
+        return source_short_path.replace('site-packages', sp_dir, 1)
+    elif source_short_path.startswith('python-scripts/'):
+        bin_dir = 'bin'
+        return source_short_path.replace('python-scripts', bin_dir, 1)
+    else:
+        return source_short_path
+
+
+def compile_pyc(python_exe_full_path, py_full_path, pyc_full_path):
+    """
+    Modified version of conda/gateways/disk/create.py:compile_pyc at
+    be8c08c083f4d5e05b06bd2689d2cd0d410c2ffe.
+
+    Modification included:
+        * removed check that the pyc file already exists prior to compilation
+        * changed log.trace -> log.info (log.trace not supported)
+
+    """
+    command = [python_exe_full_path, '-Wi', '-m', 'py_compile', py_full_path]
+    subprocess.call(command)
+
+    if not isfile(pyc_full_path):
+        log.info('{} failed to compile correctly'.format(pyc_full_path))
```

Comment:
@lbdreyer The original log message was kinda more helpful i.e.
```python
message = """
        pyc file failed to compile successfully
          python_exe_full_path: %()s\n
          py_full_path: %()s\n
          pyc_full_path: %()s\n
        """
        log.info(message, python_exe_full_path, py_full_path, pyc_full_path)
```
Did we want to add more detail here incase it barfs?